crates/lib/src/stdlib/tags/render_tag.rs

```diff
@@ -0,0 +1,474 @@
+                if !vars.is_empty() {
+                    for (id, val) in vars {
+                        let value = val
+                            .try_evaluate(runtime)
+                            .ok_or_else(|| Error::with_msg("failed to evaluate value"))?;
+
+                        pass_through.insert(id.as_ref(), value);
+                    }
+                }
+
+                let scope = SandboxedStackFrame::new(runtime, &pass_through);
+
+                let partial = scope
+                    .partials()
+                    .get(&name)
+                    .trace_with(|| format!("{{% render {} %}}", self.partial).into())?;
+
+                partial
+                    .render_to(writer, &scope)
+                    .trace_with(|| format!("{{% render {} %}}", self.partial).into())
+                    .context_key_with(|| self.partial.to_string().into())
+                    .value_with(|| name.to_string().into())?;
+            }
+            ForOrWith::For(range, var_name) => {
+                let range = range
+                    .evaluate(runtime)
+                    .trace_with(|| format!("{{% render {} %}}", self.partial).into())?;
+                let array = range.evaluate()?;
+
+                if !array.is_empty() {
+                    let len = array.len();
+                    for (i, v) in array.into_iter().enumerate() {
+                        let forloop = ForloopObject::new(i, len);
+                        let mut root = std::collections::HashMap::<
+                            liquid_core::model::KStringRef<'_>,
+                            &dyn ValueView,
+                        >::new();
+                        root.insert("foorloop".into(), &forloop);
+                        root.insert(var_name.as_ref(), &v);
+
+                        let scope = SandboxedStackFrame::new(runtime, &root);
+
+                        let partial = scope
+                            .partials()
+                            .get(&name)
+                            .trace_with(|| format!("{{% render {} %}}", self.partial).into())?;
+
+                        partial
+                            .render_to(writer, &scope)
+                            .trace_with(|| format!("{{% render {} %}}", self.partial).into())
+                            .context_key("index")
+                            .value_with(|| format!("{}", i + 1).into())?;
+
+                        // given that we're at the end of the loop body
+                        // already, dealing with a `continue` signal is just
+                        // clearing the interrupt and carrying on as normal. A
+                        // `break` requires some special handling, though.
+                        let current_interrupt =
+                            scope.registers().get_mut::<InterruptRegister>().reset();
+                        if let Some(Interrupt::Break) = current_interrupt {
+                            break;
```

Comment:
Do we need any tests for this?

Reply:
This section is pretty much an exact copy from [here](https://github.com/cobalt-org/liquid-rust/blob/master/crates/lib/src/stdlib/blocks/for_block.rs#L212), in this project. I can't see any specific tests for it there.